tests/default/indices/upgrade.yaml

```diff
@@ -0,0 +1,57 @@
+$schema: ../../../json_schemas/test_story.schema.yaml
+
+description: Test upgrading index using both GET and POST methods.
+epilogues:
+  - path: /movies
+    method: DELETE
+    status: [200, 404]
+prologues:
+  - path: /_bulk
+    method: POST
+    parameters:
+      refresh: true
+    request:
+      content_type: application/x-ndjson
+      payload:
+        - {create: {_index: movies, _id: movie1}}
+        - {director: Bennett Miller, title: The Cruise, year: 1998}
+        - {create: {_index: movies, _id: movie2}}
+        - {director: Nicolas Winding Refn, title: Drive, year: 1960}
+chapters:
+  - synopsis: Trigger index upgrade (POST).
+    path: /{index}/_upgrade
+    method: POST
+    parameters:
+      index: [movies]
+    request:
+      payload:
+        allow_no_indices: true
+        expand_wildcards: open
+        ignore_unavailable: true
+    response:
+      status: 200
+
+  - synopsis: Check index upgrade status (GET).
+    path: /{index}/_upgrade
+    method: GET
+    parameters:
+      index: [movies]
+    response:
+      status: 200
+
+  - synopsis: Trigger upgrade (POST).
```

Comment:
Note that this is not an `/{index}` API and should go in its own `tests/default/_core/upgrade.yaml`. I'll merge now, but appreciate it being split in a future PR.